File: app/src/main/java/ivankuo/com/itbon2018/di/ViewModelModule.java
package ivankuo.com.itbon2018.di;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import ivankuo.com.itbon2018.ui.RepoViewModel;
import ivankuo.com.itbon2018.viewmodel.GithubViewModelFactory;

@Module
abstract class ViewModelModule {
    @Binds
    @IntoMap
    @ViewModelKey(RepoViewModel.class)
    abstract ViewModel bindRepoViewModel(RepoViewModel repoViewModel);

    @Binds
    abstract ViewModelProvider.Factory bindViewModelFactory(GithubViewModelFactory factory);
}

File: app/src/main/java/ivankuo/com/itbon2018/di/ViewModelModule.kt
package ivankuo.com.itbon2018.di

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import ivankuo.com.itbon2018.ui.RepoViewModel
import ivankuo.com.itbon2018.viewmodel.GithubViewModelFactory

@Module
abstract class ViewModelModule {
    @Binds
    @IntoMap
    @ViewModelKey(RepoViewModel::class)
    abstract fun bindRepoViewModel(repoViewModel: RepoViewModel): ViewModel

    @Binds
    abstract fun bindViewModelFactory(factory: GithubViewModelFactory): ViewModelProvider.Factory
}